### PR TITLE
allow for more telescopic handling

### LINKED ISSUE #23119: telescopic opportunity being missed in summation?
`apart` is being used in the summation methods, but for some reason the infinite sum is missing the optimization that should result.
```python
>>> Sum(1/((5*i + 2)*(5*(i + 1) + 2)), (i, 0, oo)).doit()
gamma(12/5)/(14*gamma(7/5))
>>> Sum(apart(1/((5*i + 2)*(5*(i + 1) + 2))), (i, 0, oo)).doit()
1/10
```
The first answer is not wrong: it simplifies to 1/10 but should not need simplification.

### PR DESCRIPTION
**Pull request by smichr**
#### References to other Issues or PRs

fixes #23119 by allowing `apart` to be used on Add and Mul

#### Brief description of what is fixed or changed

The inline comments show how `apart` can lead to expressions recognizable as solvable by telescopic methods, whether starting as an Add or Mul.

#### Other comments

In master, one of the added tests gives this:
```python
>>> eq = 1/((5*n + 2)*(5*(n + 1) + 2))
>>> nz = symbols('nz', nonzero=True)
>>> v = Sum(eq.subs(5, nz), (n, 0, oo)).doit()
>>> v
Piecewise((lerchphi(1, 1, 2/nz)/(5*nz) - lerchphi(1, 1, 7/nz)/(5*nz), (nz + 2)/nz + (nz + 7)/nz - 9/nz > 1),
 (Sum(1/(n**2*nz**2 + 9*n*nz + 14), (n, 0, oo)), True))
>>> v.subs(nz,5)
-lerchphi(1, 1, 7/5)/25 + lerchphi(1, 1, 2/5)/25
>>> _.simplify()
-lerchphi(1, 1, 7/5)/25 + lerchphi(1, 1, 2/5)/25
>>> _.n()
oo
```
In this PR, the sum evaluation completes successfully on the lsum and rsum step. But I am still concerned that if that should fail the code that generated the `lerchphi` solution might still be activated and give an incorrect result.

#### Release Notes

* concrete
    * more summation are now recognized as telescopic sums

### CODE REVIEW

File: sympy/concrete/tests/test_sums_products.py [hunk 1]
```diff
@@ -542,7 +542,12 @@ def test_telescopic_sums():
     assert telescopic(1/m, -m/(1 + m), (m, n - 1, n)) == \
         telescopic(1/k, -k/(1 + k), (k, n - 1, n))
 
-    assert Sum(1/x/(x - 1), (x, a, b)).doit() == -((a - b - 1)/(b*(a - 1)))
+    assert Sum(1/x/(x - 1), (x, a, b)).doit() == 1/(a - 1) - 1/b
+    eq = 1/((5*n + 2)*(5*(n + 1) + 2))
+    assert Sum(eq, (n, 0, oo)).doit() == S(1)/10
+    nz = symbols('nz', nonzero=True)
+    v = Sum(eq.subs(5, nz), (n, 0, oo)).doit()
```
Comment (smichr) [inline sympy/concrete/tests/test_sums_products.py:549]: Note:
```python
>>> v.subs(nz,5)
Sum(1/(5*n + 2), (n, 0, oo))/5 - Sum(1/(5*n + 7), (n, 0, oo))/5
>>> s = _
>>> assert s.doit() is S.NaN
>>> s.simplify(doit=False)
Sum(-1/(5*n + 7) + 1/(5*n + 2), (n, 0, oo))/5
>>> _.doit()
1/10
```
Those last two steps are accomplished with the call to `simplify`

File: sympy/core/tests/test_expr.py [hunk 2]
```diff
@@ -2214,3 +2215,9 @@ def test_21494():
 
 def test_Expr__eq__iterable_handling():
     assert x != range(3)
+
+
+def test_issue_23120():
+    assert (x*oo).is_Mul
```
Comment (smichr) [inline sympy/core/tests/test_expr.py:2221]: `-oo` is handled in the same loop and `zoo` already handles this case so no additional tests were added beyond the one for `oo`.

File: sympy/core/tests/test_expr.py [hunk 3]
```diff
@@ -2214,3 +2215,9 @@ def test_21494():
...
+    nz = symbols('nz', nonzero=True)
+    assert nz*oo is oo
```
Comment (jksuom) [inline sympy/core/tests/test_expr.py:2223]: Shouldn't `nz` be positive?

**Comment by smichr** [comment]
Thanks, @jksuom, for the review. I restored the Mul behavior which was correct but added behavior in the Add.flatten to ignore finite terms relative to terms that are infinite or have infinite factors (for reasons given in the commit note).

**Pull request merged**